### ISSUE TITLE
Investigate removing most use of redux in the front end

### ISSUE THREAD

**Issue by AlanGreene** [tektoncd/dashboard#1842]
Redux is not really necessary for most of our state since most slices of state are only used on a small number of pages in very specific contexts. Very little state is common to the bulk of the application, we could probably reduce this to just the install properties, extensions, namespaces, and maybe one or two other slices.

Our current use leads to performance issues when a page has been left open for a long time connected to a busy cluster, e.g. TaskRun resources are only required on 3 pages (TaskRuns list, TaskRun details, and PipelineRun details), but all pages bear the cost of processing and storing at least some TaskRun data, i.e. when we receive a websocket message containing a TaskRun update even when not currently viewing anything TaskRun-related. We store this data indefinitely in memory, until we either receive a delete update, or the page is refreshed / closed, leading to excessive memory usage and gradual slow down as the volume of data / updates increases.

Investigate alternative approaches to managing what is essentially server state mirrored in the frontend. One interesting option to consider would be [React Query](https://react-query.tanstack.com/docs/overview. There may also be other packages that are more suited to our use cases, we should be able to run some small scale experiments to compare them.

**Comment by AlanGreene** [comment]
This could also simplify our efforts if we want to consume historical data from https://github.com/tektoncd/results in future. Something to consider / discuss...

**Comment by AlanGreene** [comment]
This should be one of the next big changes we make as it enables a number of interesting possibilities and integrations.
/lifecycle frozen

**Comment by AlanGreene** [comment]
This is currently blocked by #2000 which is in progress.

**Comment by AlanGreene** [comment]
Tracking progress of remaining work…

Resources to migrate to react-query:
- [ ] install properties 🚧 #2091
- [ ] ClusterInterceptors 🚧 
- [ ] ClusterTasks
- [ ] ClusterTriggerBindings 🚧 
- [ ] Conditions 🚧 
- [ ] EventListeners 🚧 
- [ ] Extensions
- [ ] Namespaces
- [ ] PipelineResources 🚧 
- [ ] PipelineRuns
- [ ] Pipelines
- [ ] ServiceAccounts 🚧 
- [ ] TaskRuns 🚧 
- [ ] Tasks
- [ ] Triggers 🚧 
- [ ] TriggerBindings 🚧 
- [ ] TriggerTemplates 🚧 

Actions, reducers, selectors, and associated tests can be deleted for each of these.
Once all of the above are done we can also remove a number of utils that are no longer needed.

- [ ] actionCreators
- [ ] reducerCreators
- [ ] selectorCreators
- [ ] constants / utils

I think this leaves just two slices of state in redux:

- [ ] locale - we can handle this locally in App.js, no need for anything more complex
- [ ] notifications (websocket connection status)
  We should be able to handle this via `queryClient.invalidateQueries()` or similar to mark all queries as stale and automatically refetch any active queries, maybe via our existing `useWebsocketReconnected` hook or more likely replacing it with something more centralised. This should probably be done last.

**Comment by briangleeson** [comment]
Planning to look at `Extensions` refactor from the above list

**Comment by AlanGreene** [comment]
#2154 open to remove last remaining use of redux, once that's merged we can clean up the remaining websocket-related code.

**Comment by AlanGreene** [comment]
All redux-related code is gone 🎉 

Migration to the new websocket model is being done in https://github.com/tektoncd/dashboard/pull/2157 in preparation for https://github.com/tektoncd/dashboard/issues/2103
